Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -115,7 +115,13 @@ class TopSitesPanel: UIViewController {
     func notificationReceived(notification: NSNotification) {
         switch notification.name {
         case NotificationFirefoxAccountChanged, NotificationProfileDidFinishSyncing, NotificationPrivateDataClearedHistory, NotificationDynamicFontChanged:
-            refreshTopSites(maxFrecencyLimit)
+            // Only reload top sites if there the cache is dirty since the finish syncing 
```

Comment:
when we receive `NotificationDynamicFontChanged` we will want to re-render regardless of whether the cached sites have changed. I think you need to move this one case to a separate option that doesn't do the check.